src/oop/special_class_method/medium/q2/MainTest.java

```diff
@@ -0,0 +1,236 @@
+    public void correctGetBrightnessMethod(double value) throws Throwable {
+        Object[][] arguments = {
+                {value, double.class}
+        };
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(getAttributeMethodName1, null), String.format(incorrectMethodDefinition, getAttributeMethodName1, className));
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"), String.format(incorrectModifierMessage, getAttributeMethodName1, className));
+        assertTrue(testClass.hasReturnType(getAttributeMethodName1, null, double.class), String.format(incorrectReturnType, getAttributeMethodName1, className));
+        Object testInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+        _assertEquals(value, getMethodOutput, 0.0001,
+                "Your " + getAttributeMethodName1 + " method does not return the value of the " + attributeName1 + " attribute.");
+    }
+
+    private static Stream<Arguments> twoDoubleInputProvider() {
+        return Stream.of(
+                Arguments.of(5.4, 6.4),
+                Arguments.of(654.3, 54.2),
+                Arguments.of(43.3, 43.2),
+                Arguments.of(432.2, 899.8)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoDoubleInputProvider")
+    public void correctSetBrightnessMethod(double initialValue, double updatedValue) throws Throwable {
+        Object[][] instantiationArguments = {
+                {initialValue, double.class}
+        };
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(setAttributeMethodName1, new Class<?>[]{double.class}), String.format(incorrectMethodDefinition, getAttributeMethodName1, className));
+        assertTrue(testClass.hasModifier(setAttributeMethodName1, new Class<?>[]{double.class}, "public"), String.format(incorrectModifierMessage, getAttributeMethodName1, className));
+        assertTrue(testClass.hasReturnType(setAttributeMethodName1, new Class<?>[]{double.class}, Void.TYPE), String.format(incorrectReturnType, getAttributeMethodName1, className));
+        Object testInstance = testClass.createInstance(instantiationArguments);
```

Comment:
I don't think that the assert should check the set and then have the message use the get name here.